src/app/webui/components/ServersPanel.tsx

```diff
@@ -0,0 +1,319 @@
+'use client';
+
+import React, { useState, useEffect, useCallback } from 'react';
+import { Button } from './ui/button';
+import { X, PlusCircle, Server, ListChecks, ChevronRight, RefreshCw, AlertTriangle, ChevronDown, Terminal } from 'lucide-react';
+import { cn } from '@/lib/utils';
+import Link from 'next/link';
+
+// --- Interfaces copied/adapted from PlaygroundView.tsx ---
+interface JsonSchemaProperty {
+  type: 'string' | 'number' | 'integer' | 'boolean' | 'object' | 'array';
+  description?: string;
+  default?: any;
+  enum?: Array<string | number | boolean>;
+  format?: string;
+}
+
+interface JsonSchema {
+  type?: 'object';
+  properties?: Record<string, JsonSchemaProperty>;
+  required?: string[];
+}
+
+interface McpServer {
+  id: string;
+  name: string;
+  status: 'connected' | 'disconnected' | 'error' | 'unknown';
+  // other relevant details
+}
+
+interface McpTool {
+  id: string;
+  name: string;
+  description?: string;
+  inputSchema?: JsonSchema | null;
+  // other relevant details
+}
+// --- End of Interfaces ---
+
+interface ServersPanelProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onOpenConnectModal: () => void;
+  variant?: 'overlay' | 'inline';
+}
```

Comment:
let's not use duplicate types unless there is a good reason to - store types in `types.ts` or some common place and reuse them everywhere

Reply:
if the types already exist in core, use those. if they don't exist, create re-usable types in `types.ts`